lib/Myriad/Bootstrap.pm

```diff
@@ -171,50 +232,41 @@ sub boot {
                 die "excessive module loading detected: $loaded_modules" if $loaded_modules;
             }
 
-            my $stop = 0;
             local $SIG{HUP} = sub {
                 say "$$ - HUP detected in parent";
                 kill 3, $pid;
-    #           $stop = 1;
             };
 
-            # Build up any output from the child process
-            my $input = '';
+            print $child_pipe "Parent active\r\n";
 
-            my $rin = my $win = '';
-            vec($rin, fileno($child_pipe), 1) = 1;
-            my $ein = $rin | $win;
             ACTIVE:
-            while(!$stop) {
-                say "$$ Parent loop cycle";
-                die $! unless defined(my $nfound = select my $rout = $rin, my $wout = $win, my $eout = $ein, 5);
-                if($nfound) {
-    #               say "$$ Child has something to report ($nfound): $rout, $wout, $eout";
-                    my $rslt = sysread $child_pipe, my $buf, 4096;
-                    last ACTIVE unless $rslt;
-                    $input .= $buf;
-                    while($input =~ s/^[\r\n]*(.*)[\r\n]+//) {
-                        say "Received from child: $1";
+            while (1) {
+
+                last ACTIVE unless check_messages_in_pipe($inotify_child_pipe, sub {
+                    say "$$ - File has been detected reloading..";
+                    kill 3, $pid;
+                    next MAIN;
+                });
+
+                last ACTIVE unless check_messages_in_pipe($child_pipe, sub {
+                    my $module = shift;
+                    print $inotify_child_pipe "$module\r\n";
+                });
+
+                for my $child_pid (@children_pids) {
+                    if(my $exit = waitpid $pid, $constant{WNOHANG}) {
+                        say "$$ Exit was $exit";
+                        # stop the other processes
+                        kill 1, $_ for grep {$_ eq $child_pid} @children_pids;
+                        last MAIN;
                     }
                 }
             }
-            if(my $exit = waitpid $pid, 0) {
-                say "$$ Exit was $exit";
-                last MAIN;
-            } else {
-                say "$$ No exit code yet";
-            }
+
             say "$$ - Done";
             exit 0;
         } else {
             say "$$ - Child with parent " . $parent_pid;
```

Comment:
this kid has access to all 4 pipe ends. Is that intended? Better to close what you don't need.